### PR TITLE
Fix region exception when AWS_DEFAULT_REGION not set

### LINKED ISSUE #136: exception raised on OS environment setting 'AWS_DEFAULT_REGION'
**Describe the bug**
unable to run taskcat using latest version 2018.913.194818
getting below errors.
Traceback (most recent call last):
  File "/usr/local/bin/taskcat", line 72, in <module>
    main()
  File "/usr/local/bin/taskcat", line 60, in main
    tcat_instance.stage_in_s3(taskcat_cfg)
  File "/usr/local/lib/python3.6/site-packages/taskcat/stacker.py", line 415, in stage_in_s3
    s3_client = self._boto_client.get('s3', region=self.get_default_region(), s3v4=True)
  File "/usr/local/lib/python3.6/site-packages/taskcat/utils.py", line 118, in get
    region = os.environ['AWS_DEFAULT_REGION']
  File "/usr/lib64/python3.6/os.py", line 669, in __getitem__
    raise KeyError(key) from None
KeyError: 'AWS_DEFAULT_REGION'

### PR DESCRIPTION
**Pull request by jaymccon**
## Overview

Fixes #136

## Testing/Steps taken to ensure quality

Ran `python setup.py test`

invoked taskcat using the `--profile` argument

### Notes

Also threw in a [commit](https://github.com/aws-quickstart/taskcat/commit/1d849626c8d900296f33aab6aba75ee190701530) that fixes a bug that prevented taskcat from exiting cleanly when `exit0` or `exit1` were called.

## Testing Instructions

run taskcat using the `--profile` argument and AWS_DEFAULT_REGION not set as an environment variable.

### CODE REVIEW

File: taskcat/client_factory.py [hunk 1]
```diff
@@ -66,6 +66,37 @@ def __init__(self, logger=None, loglevel='error', botolevel='error', aws_access_
         self.put_credential_set('default', aws_access_key_id, aws_secret_access_key, aws_session_token, profile_name)
         return
 
+    def get_default_region(self, aws_access_key_id, aws_secret_access_key, aws_session_token, profile_name):
```
Comment (jaymccon) [inline taskcat/client_factory.py]: getting the default region properly turns out to be a bit complex

File: taskcat/stacker.py [hunk 2]
```diff
@@ -1755,6 +1755,8 @@ def aws_api_init(self, args):
             either profile name, access key and secret key or none.
         """
         print('\n')
+
+        self.set_default_region = ClientFactory().get_default_region(args.aws_access_key, args.aws_secret_key, None, args.boto_profile)
```
Comment (jaymccon) [inline taskcat/stacker.py]: you'll be pleased to see I'm reusing this everywhere, not just 2/3

**Comment by jaymccon** [comment]
@agamzn ready for another look

File: taskcat/stacker.py (hunk 2 above)
Comment (agamzn) [reply taskcat/stacker.py]: I don't understand the use case for overriding the function. Did you intead this to read either of these? 
-  `self.region = ClientFactory().get_default_region(args.aws_access_key, args.aws_secret_key, None, args.boto_profile)`
- `self.set_default_region(region=ClientFactory().get_default_region(args.aws_access_key, args.aws_secret_key, None, args.boto_profile)` here?

**Comment by jaymccon** [comment]
@agamzn nice catch, thanks, I've fixed it

**Pull request merged**